cobc/codegen.c

```diff
@@ -7068,6 +7090,74 @@ output_field_constant (cb_tree x, int n, const char *flagname)
 	output_newline ();
 }
 
+static void
+output_exception_handling (struct cb_call *p)
+{
+	if (p->stmt1) {
+		output_line ("cob_glob_ptr->cob_stmt_exception = 1;");
+		output_line ("COB_RESET_EXCEPTION(0);");
+	} else {
+		output_line ("cob_glob_ptr->cob_stmt_exception = 0;");
+	}
+
+	output_line ("if ((cob_glob_ptr->cob_exception_code & 0xff00) != 0) ");
+	output_block_open ();
+
+	if (p->stmt1) {
+		output_stmt (p->stmt1);
+	} else if (p->stmt2) {
+		output_stmt (p->stmt2);
+	}
+	output_block_close ();
+	output_line ("COB_RESET_EXCEPTION(0);");
+}
+
+static void
+output_java_call (struct cb_call *p)
+{
+	char		*full_name, *class_and_method_name, *last_dot, *c;
+	const char	*class_name, *method_name;
+	char		mangled[COB_NORMAL_BUFF];
+
+	/* TODO: move that back into cb_check_conformance (*) */
```

Comment:
@xevor11 I think you have started moving this check and the one below back into `cb_check_conformance`; so I'll stop pushing changes to this branch to let you advance with that.